wandb/apis/public.py

```diff
@@ -4441,3 +4844,128 @@ def convert_objects(self):
 
     def __repr__(self):
         return "<ArtifactFiles {} ({})>".format("/".join(self.artifact.path), len(self))
+
+
+class Job(Media):
+    _log_type = "job"
+
+    _name: str
+    _input_types: Type
+    _output_types: Type
+    _entity: str
+    _project: str
+    _entrypoint: List[str]
+
+    def __init__(self, client: Api, name, path: str = None) -> None:
+
+        self._job_artifact = client.artifact(name, type="job")
+        if path:
+            self._fpath = path
+            self._job_artifact.download(root=path)
+        else:
+            self._fpath = self._job_artifact.download()
```

Comment:
How are you guaranteed to have the corresponding artifact?

Reply:
I'll wrap this in a try and raise an appropriate error